commit 7cfb0435330364f90f274a26ecdc5f47f738498c
Author: Thomas Gleixner <tglx@linutronix.de>
Date:   Wed Sep 3 21:37:24 2008 +0000

    HPET: make minimum reprogramming delta useful
    
    The minimum reprogramming delta was hardcoded in HPET ticks,
    which is stupid as it does not work with faster running HPETs.
    The C1E idle patches made this prominent on AMD/RS690 chipsets,
    where the HPET runs with 25MHz. Set it to 5us which seems to be
    a reasonable value and fixes the problems on the bug reporters
    machines. We have a further sanity check now in the clock events,
    which increases the delta when it is not sufficient.
    
    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
    Tested-by: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>
    Tested-by: Dmitry Nezhevenko <dion@inhex.net>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

---
 arch/x86/kernel/hpet.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.24.7-rt27/arch/x86/kernel/hpet.c
===================================================================
--- linux-2.6.24.7-rt27.orig/arch/x86/kernel/hpet.c	2009-02-07 23:59:34.000000000 -0500
+++ linux-2.6.24.7-rt27/arch/x86/kernel/hpet.c	2009-02-08 00:05:11.000000000 -0500
@@ -221,8 +221,8 @@ static void hpet_legacy_clockevent_regis
 	/* Calculate the min / max delta */
 	hpet_clockevent.max_delta_ns = clockevent_delta2ns(0x7FFFFFFF,
 							   &hpet_clockevent);
-	hpet_clockevent.min_delta_ns = clockevent_delta2ns(0x30,
-							   &hpet_clockevent);
+	/* 5 usec minimum reprogramming delta. */
+	hpet_clockevent.min_delta_ns = 5000;
 
 	/*
 	 * Start hpet with the boot cpu mask and make it
